src/root.jsx

```diff
@@ -5,45 +5,23 @@ import DataTable from "./data-table/index.jsx";
 import { LinkedScrollWrapper, LinkedScrollSection } from "./linked-scroll";
 
 class Root extends React.PureComponent {
-  constructor (props) {
-    super(props);
-    this.onDataTableRefSet = this.onDataTableRefSet.bind(this);
-    this.renderedTableWidth = 0;
-  }
-
-  componentDidUpdate () {
-    let tableWidth;
-    if (this.dataTableRef) {
-      tableWidth = this.dataTableRef.getBoundingClientRect().width;
-      if (this.renderedTableWidth !== tableWidth) {
-        this.forceUpdate();
-      }
-    }
-  }
-
-  onDataTableRefSet (element) {
-    this.dataTableRef = element;
-    this.forceUpdate();
-  }
-
   render () {
     const { editmodeClass, component, state } = this.props;
-    const { data, general, styling, error } = state;
+    const { data, general, styling, error, element } = state;
 
     // Determine cell- and column separator width
     let cellWidth = '0px';
     let columnSeparatorWidth = '';
-    if (this.dataTableRef && !error) {
-      const tableWidth = this.dataTableRef.getBoundingClientRect().width;
-      this.renderedTableWidth = tableWidth;
+    if (!error && element) {
+      const tableWidth = element.getBoundingClientRect().width;
 
       if (general.cellWidth) {
         cellWidth = general.cellWidth;
         if (general.useColumnSeparator) {
           columnSeparatorWidth = '8px';
         }
       } else {
-        const headerMarginRight = 8;
+        const headerMarginRight = 8 + 230 + 20;
```

Comment:
might be good do declare those values 230 and 20 as const to explain what the values are coming from...

Reply:
hmm, I thought I added a comment. It is basically magic numbers, the 230 is the width of the first dim columnm, the rest is just various paddings.